source/agora/test/EnrollmentManager.d

```diff
@@ -26,48 +26,80 @@ import agora.consensus.EnrollmentManager;
 import agora.consensus.Genesis;
 import agora.test.Base;
 
-/// test for  enrollment process & revealing a pre-image periodically
-unittest
-{
-    auto network = makeTestNetwork(TestConf.init);
-    network.start();
-    scope(exit) network.shutdown();
-    scope(failure) network.printLogs();
-    network.waitForDiscovery();
+import std.typecons: Tuple;
 
+version(unittest)
+Tuple!(Transaction[], Enrollment[]) enrollValidators (T : TestAPIManager = TestAPIManager)(T network, bool[] not_enrolled_validators)
+{
+    assert(network.nodes.length == not_enrolled_validators.length);
     auto nodes = network.clients;
     auto node_1 = nodes[0];
-    auto node_2 = nodes[1];
-
-    // make transactions which have UTXOs for the node.
     auto gen_key_pair = getGenesisKeyPair();
-    auto pubkey_1 = node_1.getPublicKey();
 
     Transaction[] txs;
-    foreach (idx; 0 .. Block.TxsInBlock)
+    int tx_idx = 0;
+    foreach (idx; 0 .. not_enrolled_validators.length)
```

Comment:
Style note, this would be much better expressed as:
```suggestion
    foreach (idx, not_enrolled; not_enrolled_validators)
```

But remember: Double negatives are bad. Instead of "!not_enrolled", we should make this `enrolled`.
But that is just for an array of `bool`. In practice, a better way to do this would be to pass a range that has been `filter`ed out of the values we want to exclude.